Lib/asynchat.py

```diff
@@ -122,7 +122,7 @@ def handle_read(self):
             return
 
         if isinstance(data, str) and self.use_encoding:
-            data = bytes(str, self.encoding)
+            data = bytes(data, self.encoding)
```

Comment:
Arguments for "yes, just remove the code":
1. Code doesn't seem to have ever been triggered in reality
2. If a socket (or socket-like thing) returns a unicode string, that smells like a bug with the socket
3. Implied zens: Errors should never pass silently. Unless explicitly silenced.

Arguments for "no, keep the code and fix it":
1. This was an explicit fix/silencing of an error
2. The intent was to "do the right thing", the existing code can only raise an exception if the clause hits (existing code doesn't do the right thing)
3. Deleting the code will only induce an exception if this case comes up in the future (deleting the code may produce a questionable "right thing")
4. Trivial fix

The real question is whether we want to raise an exception in the case where a socket returns a unicode string, or whether we want to transparently transform the data to bytes with the connection's encoding.

Short version: I think it should be fixed, because that was my intent when I merged the patch 9 years ago.